radicle-seed/src/lib.rs

```diff
@@ -0,0 +1,271 @@
+// This file is part of radicle-link
+// <https://github.com/radicle-dev/radicle-link>
+//
+// Copyright (C) 2019-2020 The Radicle Team <dev@radicle.xyz>
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 or
+// later as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![allow(clippy::single_match)]
+#[macro_use]
+extern crate async_trait;
+
+use futures::stream::StreamExt;
+use std::{collections::HashSet, io, net::SocketAddr, path::PathBuf, vec};
+use thiserror::Error;
+
+use radicle_keystore::sign::ed25519;
+
+use librad::{
+    git,
+    keys,
+    meta::project,
+    net::{
+        discovery,
+        gossip,
+        gossip::types::PeerInfo,
+        peer::{self, PeerApi, PeerConfig},
+        protocol::ProtocolEvent,
+    },
+    paths,
+    peer::PeerId,
+    uri::RadUrn,
+};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Storage(#[from] git::storage::Error),
+
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    Bootstrap(#[from] peer::BootstrapError),
+
+    #[error(transparent)]
+    Accept(#[from] peer::AcceptError),
+}
+
+#[derive(Clone)]
+pub struct Signer {
+    key: keys::SecretKey,
+}
+
+impl Signer {
+    pub fn new<R: io::Read>(mut r: R) -> Result<Self, io::Error> {
+        use radicle_keystore::SecretKeyExt;
```

Comment:
Why is this `use` inlined?

Reply:
No particular reason except that it's only used here and is kind of esoteric